File: examples/bench/cpp/oop.cpp
#include <cstdio>
struct Point {
    long long x, y;
    Point(long long x_, long long y_) : x(x_), y(y_) {}
    void add(const Point& o) { x += o.x; y += o.y; }
};
int main() {
    long long total = 0;
    for (long long i = 0; i < 5000000; i++) {
        Point a(i, i + 1);
        Point b(1, 1);
        a.add(b);
        total += a.x + a.y;
    }
    std::printf("%lld\n", total);
}

File: examples/bench/py/oop.py
class Point:
    def __init__(self, x, y):
        self.x = x
        self.y = y
    def mag_sq(self):
        return self.x * self.x + self.y * self.y

total = 0
i = 0
while i < 5000000:
    p = Point(i, i + 1)
    total += p.mag_sq()
    i += 1
print(total)